### PR TITLE
Clean-up `typing_extensions` -- `typing_extensions`

### PR DESCRIPTION
**Pull request by AA-Turner**
xref #892 
@srittau 

Actual `typing_extensions` PR. Unchanged from original.

A

### CODE REVIEW

File: typing_extensions/src_py3/typing_extensions.py [hunk 1]
```diff
@@ -1698,7 +1046,7 @@ def __init__(cls, name, bases, ns, total=True):
             # In Python 3.4 and 3.5 the __init__ method also needs to support the
             # keyword arguments.
             # See https://www.python.org/dev/peps/pep-0487/#implementation-details
```
Comment (srittau) [inline typing_extensions/src_py3/typing_extensions.py:1048]: This comment seems obsolete.

File: typing_extensions/src_py3/typing_extensions.py [hunk 2]
```diff
@@ -1780,7 +1128,10 @@ class Point2D(TypedDict):
     # Not exported and not a public API, but needed for get_origin() and get_args()
     # to work.
     _AnnotatedAlias = typing._AnnotatedAlias
+# 3.7-3.8
 elif PEP_560:
+    import operator
```
Comment (srittau) [inline typing_extensions/src_py3/typing_extensions.py:1133]: Nit: While this import is only used for this branch, I suggest to import at the top for better overview.

**Review by srittau — CHANGES_REQUESTED** [review]
Thanks!

You (sensibly) removed a few non-underscored, but also not exported via `__all__` attributes. As this is technically an API change (even though it's not exported via `__all__`), could you add these removals to `typing_extensions/CHANGELOG`:

* `TypingMeta`
* `OLD_GENERICS`
* `SUBS_TREE`
* `HAVE_ANNOTATED`
* `HAVE_PROTOCOLS`

Also could you mention in the Changelog that starting with 4.0.0 support for Python 2 and <3.6 has been dropped?

A few more remarks below.

File: typing_extensions/src_py3/typing_extensions.py (hunk 2 above)
Comment (AA-Turner) [reply typing_extensions/src_py3/typing_extensions.py]: Done

File: typing_extensions/src_py3/typing_extensions.py (hunk 1 above)
Comment (AA-Turner) [reply typing_extensions/src_py3/typing_extensions.py]: Done

File: typing_extensions/src_py3/typing_extensions.py [hunk 3]
```diff
@@ -155,27 +78,16 @@ def _check_methods_in_mro(C, *methods):
...
-# Annotated relies on substitution trees of pep 560. It will not work for
-# versions of typing older than 3.5.3
-HAVE_ANNOTATED = PEP_560 or SUBS_TREE
-
 if PEP_560:
     __all__.extend(["get_args", "get_origin", "get_type_hints"])
 
-if HAVE_ANNOTATED:
-    __all__.append("Annotated")
+__all__.extend(['Annotated', 'Protocol', 'runtime', 'runtime_checkable'])
```
[the remark this answers is not in the document]
Comment (AA-Turner) [reply typing_extensions/src_py3/typing_extensions.py]: Done

**Comment by AA-Turner** [comment]
Sorry, forgot to add a changelog entry earlier -- have now done so.

**Review by srittau — APPROVED** [review]
Thanks again. For future reference: Please don't force push as it makes it harder to review changes.

**Comment by AA-Turner** [comment]
Sorry - I had to rebase onto `master` to pick up the changelog -- is it better to merge in `HEAD` instead of rebase + force push?
A

**Comment by srittau** [comment]
Yes, we squash merge anyway, so rebasing should not be necessary.

**Pull request merged**